ducktape/mark/_mark.py

```diff
@@ -134,6 +134,47 @@ def __eq__(self, other):
         return super(Matrix, self).__eq__(other) and self.injected_args == other.injected_args
 
 
+class Defaults(Mark):
+    """Parametrize with a default matrix of arguments on existing parametrizations.
+    Assume each values in self.injected_args is iterable
+    """
+
+    def __init__(self, **kwargs):
+        self.injected_args = kwargs
+        for k in self.injected_args:
+            try:
+                iter(self.injected_args[k])
+            except TypeError, te:
+                raise DucktapeError("Expected all values in @defaults decorator to be iterable: " + str(te))
+
+    @property
+    def name(self):
+        return "DEFAULTS"
+
+    def apply(self, seed_context, context_list):
+        new_context_list = []
+        if context_list:
+            for ctx in context_list:
+                for injected_args in cartesian_product_dict(self.injected_args):
+                    if set(injected_args).intersection(set(ctx.injected_args)):
```

Comment:
Not sure I understand the logic here. If there is *any* overlap in the set of default args and the args, we basically do nothing to that parameterization? I guess I would have expected that defaults apply argument by argument.

Reply:
now I think about it, it is probably more intuitive to have this apply arg by arg, I did that initially and changed it for some reason..